src/parser.ts

```diff
@@ -498,7 +496,14 @@ export class Parser {
       propsObj,
       propsObj.valueDeclaration
     );
-    const propertiesOfProps = propsType.getProperties();
+    let propertiesOfProps = propsType.getProperties();
+
+    if (!propertiesOfProps.length && propsType.isUnionOrIntersection()) {
+      propertiesOfProps = propsType.types.reduce<ts.Symbol[]>(
+        (acc, type) => [...acc, ...type.getProperties()],
+        []
+      );
+    }
```

Comment:
Every change below this line was done by the commit hook. I only added lines 499-506

Reply:
Thanks for pointing that out :) It's really annoying but I have no idea why this is happening. It should just run `npm run prettier:base` which in turn runs `prettier --parser typescript --single-quote --trailing-comma none`.  Not sure why this should be different for different people without changing `prettier` version.